### ISSUE TITLE
Add a /network/xxx parameter with sanity check

### ISSUE THREAD

**Issue by pimato** [protofire/omen-exchange#1534]
As we're supporting other non-test networks like XDai we need to be able to handle incoming links from users who may not be on mainnet.

Add support for a /network/xxx parameter where xxx is the network ID. If the network ID specified doesn't match the detected network, show a warning to the user that they should switch.

Example URL:
`https://omen.eth.link//#/0xf9cefa4ade74834882589f5309664c35ecdeff5f/network/100`

Json file with all network information: https://chainid.network/chains.json
For more info: https://chainid.network/

Network Data:

```,
  {
    "name": "Ethereum Mainnet",
    "chainId": 1,
    "shortName": "eth",
    "chain": "ETH",
    "network": "mainnet",
    "networkId": 1,
    "nativeCurrency": {"name":"Ether","symbol":"ETH","decimals":18},
    "rpc": ["https://mainnet.infura.io/v3/${INFURA_API_KEY}","wss://mainnet.infura.io/ws/v3/${INFURA_API_KEY}","https://api.mycryptoapi.com/eth","https://cloudflare-eth.com"],
    "faucets": [],
    "infoURL": "https://ethereum.org"
  },
  {
    "name": "Ethereum Testnet Rinkeby",
    "chainId": 4,
    "shortName": "rin",
    "chain": "ETH",
    "network": "rinkeby",
    "networkId": 4,
    "nativeCurrency": {"name":"Rinkeby Ether","symbol":"RIN","decimals":18},
    "rpc": ["https://rinkeby.infura.io/v3/${INFURA_API_KEY}","wss://rinkeby.infura.io/ws/v3/${INFURA_API_KEY}"],
    "faucets": ["https://faucet.rinkeby.io"],
    "infoURL": "https://www.rinkeby.io"
  },
  {
    "name": "xDAI Chain",
    "chainId": 100,
    "shortName": "xdai",
    "chain": "XDAI",
    "network": "mainnet",
    "networkId": 100,
    "nativeCurrency": {"name":"xDAI","symbol":"xDAI","decimals":18},
    "rpc": ["https://rpc.xdaichain.com","https://xdai.poanetwork.dev","wss://rpc.xdaichain.com/wss","wss://xdai.poanetwork.dev/wss","http://xdai.poanetwork.dev","https://dai.poa.network","ws://xdai.poanetwork.dev:8546"],
    "faucets": [],
    "infoURL": "https://forum.poa.network/c/xdai-chain"
  },
  {
    "name": "POA Network Sokol",
    "chainId": 77,
    "shortName": "poa",
    "chain": "POA",
    "network": "sokol",
    "networkId": 77,
    "nativeCurrency": {"name":"POA Sokol Ether","symbol":"POA","decimals":18},
    "rpc": ["https://sokol.poa.network","wss://sokol.poa.network/wss","ws://sokol.poa.network:8546"],
    "faucets": ["https://faucet-sokol.herokuapp.com"],
    "infoURL": "https://poa.network"
  },

**Comment by auryn-macmillan** [comment]
I'd suggest that network could be passed via a URL parameter, rather than as part of the route.
`https://omen.eth.link/#/0xed6940e49cee029c4664d9794c04d5f9e5597632?network=xdai`

This way we don't break existing links.

**Comment by KadenZipfel** [comment]
The more I dig into this implementation, the more problems I run into 😭

- We have to apply this separately for each page as a result of some routing constraints, meaning a lot of code needs to be written and maintained, and if we add new pages this has to be applied to each new page.
  - Different pages have different events in which they push to history in various ways, which would all need to be updated to work correctly, meaning lots of dev time and lots of possibility bugs. 
    - This is especially a problem on the homepage, where the network path overrides the market filters and vice versa.

In general, this implementation results in a lot of unexpected effects, and requires a lot more changes than I had initially expected, of which it will be easy to miss changes and otherwise create bugs.

A much simpler solution, which will result in less bugs and maintenance would be to use a different domain. If we did this, we could have one domain for mainnet/rinkeby and one for xdai/sokol. Then if a user visits an xdai link from mainnet, or vice versa, we can pretty simply display a warning to let them know to change networks. 

Assuming that the end goal would be to create a subdomain for xDai when possible, this implementation would also make switching over much simpler.

I've registered xomen.eth in case we want to do this, and I'm happy to transfer it to dxdao.
